src/server/lsp.rs

```diff
@@ -385,7 +385,7 @@ impl LanguageServer for TypstServer {
             .map(|context| context.trigger_kind == CompletionTriggerKind::INVOKED)
             .unwrap_or(false);
         let position_encoding = self.const_config().position_encoding;
-
+        let frames = { self.document.lock().await.pages.clone() };
```

Comment:
it would be better if we can use Arc<doucment> everywhere. so we can remove heavy clones

Reply:
done